### DISCUSSION TITLE
Changing domain breaks email verification URL

### DISCUSSION THREAD

**Discussion by Bhautik0110** [zitadel/zitadel#5135] [Q&A] [answered]
### Issue
We have deployed zitadel on a temporary domain, and now we changed the zitadel domain It breaks the user email verification link. In email verification old domain URL is coming. I changed the organization domain but cannot change instance domain. It still appears on the Instance page how we can remove the old domain permanently and use the new domain.

**Comment by livio-a** [comment, marked as answer]
Hey @Bhautik0110 

The links used in emails are built with the primary instance domain.

You can manage the instance domains on self-hosted systems through the system api:
https://zitadel.com/docs/guides/integrate/access-zitadel-system-api

There you can remove the domain (https://zitadel.com/docs/apis/proto/system#removedomain) or at least change the primary domain (https://zitadel.com/docs/apis/proto/system#setprimarydomain)

**Reply by AlexanderAkulshin** [reply]
If you don't mind I would like to ask you to replace outdated links with following

1. remove domain from instance

https://zitadel.com/docs/apis/resources/system/system-service-remove-domain#removes-the-domain-of-an-instance

2. set primary domain for instance

https://zitadel.com/docs/apis/resources/system/system-service-set-primary-domain#sets-the-primary-domain-of-an-instance
